storagext/lib/src/types/storage_provider.rs

```diff
@@ -312,6 +313,29 @@ impl From<Vec<TerminationDeclaration>> for RuntimeTerminateSectorsParams {
     }
 }
 
+pub struct PartitionState {
+    pub sectors: BTreeSet<SectorNumber>,
+}
+
+pub struct DeadlineState {
+    pub partitions: BTreeMap<PartitionNumber, PartitionState>,
+}
+
+impl From<RuntimeDeadlineState> for DeadlineState {
+    fn from(value: RuntimeDeadlineState) -> Self {
+        Self {
+            partitions: BTreeMap::from_iter(value.partitions.0.iter().map(|(k, v)| {
+                (
+                    *k,
+                    PartitionState {
+                        sectors: BTreeSet::from_iter(v.sectors.0.iter().cloned()),
```

Comment:
We have ownership of `value` meaning we should have ownership of `v`, can't we move out the values using `into_iter` instead of `cloned`?